orttraining/orttraining/test/python/orttraining_run_multiple_choice.py

```diff
@@ -0,0 +1,241 @@
+            model_args.tokenizer_name if model_args.tokenizer_name else model_args.model_name_or_path,
+            cache_dir=model_args.cache_dir,
+        )
+
+        model = AutoModelForMultipleChoice.from_pretrained(
+            model_args.model_name_or_path,
+            from_tf=bool(".ckpt" in model_args.model_name_or_path),
+            config=config,
+            cache_dir=model_args.cache_dir,
+        )
+
+        # Get datasets
+        train_dataset = (
+            MultipleChoiceDataset(
+                data_dir=data_args.data_dir,
+                tokenizer=tokenizer,
+                task=data_args.task_name,
+                max_seq_length=data_args.max_seq_length,
+                overwrite_cache=data_args.overwrite_cache,
+                mode=Split.train,
+            )
+            if training_args.do_train
+            else None
+        )
+        eval_dataset = (
+            MultipleChoiceDataset(
+                data_dir=data_args.data_dir,
+                tokenizer=tokenizer,
+                task=data_args.task_name,
+                max_seq_length=data_args.max_seq_length,
+                overwrite_cache=data_args.overwrite_cache,
+                mode=Split.dev,
+            )
+            if training_args.do_eval
+            else None
+        )
+
+        def compute_metrics(p: EvalPrediction) -> Dict:
+            preds = np.argmax(p.predictions, axis=1)
+            return {"acc": simple_accuracy(preds, p.label_ids)}
+
+        if model_name.startswith('bert'):
+            model_desc = ModelDescription([
+                IODescription('input_ids', [self.train_batch_size, num_labels, data_args.max_seq_length], torch.int64, num_classes=model.config.vocab_size),
+                IODescription('attention_mask', [self.train_batch_size, num_labels, data_args.max_seq_length], torch.int64, num_classes=2),
+                IODescription('token_type_ids', [self.train_batch_size, num_labels, data_args.max_seq_length], torch.int64, num_classes=2),
+                IODescription('labels', [self.train_batch_size, num_labels], torch.int64, num_classes=num_labels)], [
+                IODescription('loss', [], torch.float32),
+                IODescription('reshaped_logits', [self.train_batch_size, num_labels], torch.float32)])
+        else:
+            model_desc = ModelDescription([
+                IODescription('input_ids', ['batch', num_labels, 'max_seq_len_in_batch'], torch.int64, num_classes=model.config.vocab_size),
+                IODescription('attention_mask', ['batch', num_labels, 'max_seq_len_in_batch'], torch.int64, num_classes=2),
+                IODescription('labels', ['batch', num_labels], torch.int64, num_classes=num_labels)], [
+                IODescription('loss', [], torch.float32),
+                IODescription('reshaped_logits', ['batch', num_labels], torch.float32)])
+
+        # Initialize the ORTTrainer within ORTTransformerTrainer
+        trainer = ORTTransformerTrainer(
```

Comment:
Why not `ORTTrainer`? What is the difference?

Reply:
this trainer wraps ORTTrainer. It also encapsulate training test datasets, training/eval looping, matrix computation. On the other hand it is too specific to transformer tasks otherwise I would suggest we adopt the approach.